src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,187 @@
+package duke.ui;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The `Ui` class handles user interaction and provides formatted output.
+ * It includes methods for greeting the user, formatting replies, displaying errors,
+ * and obtaining user input.
+ */
+public class Ui {
+
+    /**
+     * Scanner object for reading user input.
+     */
+    private final Scanner scanner;
+
+    /**
+     * Horizontal line used for formatting output.
+     */
+    private final String horizontalLine = "____________________________________________________________\n";
```

Comment:
Should these be in SCREAMING_SNAKE_CASE? Noticed this with several `final` constants.